server/bleep/src/semantic.rs

```diff
@@ -199,45 +184,73 @@ impl Semantic {
     ) -> Result<Self, SemanticError> {
         let qdrant = QdrantClient::new(Some(QdrantClientConfig::from_url(qdrant_url))).unwrap();
 
-        match qdrant.has_collection(COLLECTION_NAME).await {
+        match qdrant.has_collection(&config.collection_name).await {
             Ok(false) => {
                 let CollectionOperationResponse { result, time } = qdrant
-                    .create_collection(&collection_config())
+                    .create_collection(&CreateCollection {
+                        collection_name: config.collection_name.to_string(),
+                        vectors_config: Some(VectorsConfig {
+                            config: Some(vectors_config::Config::Params(VectorParams {
+                                size: EMBEDDING_DIM as u64,
+                                distance: Distance::Cosine.into(),
+                                ..Default::default()
+                            })),
+                        }),
+                        ..Default::default()
+                    })
                     .await
                     .unwrap();
 
                 debug!(
                     time,
                     created = result,
-                    name = COLLECTION_NAME,
+                    name = config.collection_name,
                     "created qdrant collection"
                 );
 
                 assert!(result);
+
+                qdrant
+                    .create_field_index(
+                        &config.collection_name,
+                        "repo_ref",
+                        FieldType::Text,
+                        None,
+                        None,
+                    )
+                    .await?;
+                qdrant
+                    .create_field_index(
+                        &config.collection_name,
+                        "content_hash",
+                        FieldType::Text,
+                        None,
+                        None,
+                    )
+                    .await?;
+                qdrant
+                    .create_field_index(
+                        &config.collection_name,
+                        "branches",
+                        FieldType::Text,
+                        None,
+                        None,
+                    )
+                    .await?;
+                qdrant
+                    .create_field_index(
+                        &config.collection_name,
+                        "relative_path",
+                        FieldType::Text,
+                        None,
+                        None,
+                    )
+                    .await?;
```

Comment:
Note: these were outside of the `match` because we needed to migrate previously created collections to ones with indexes. Moving this inside the `match`  may subtly cause issues with upgrades if we rely on indexes being present in the future, if we e.g. choose to extend this code block.

Maybe a better way of dealing with any potential failures on this code path is splitting this section into a new function, loop over the indexed field names, and explicitly logging the results of these operations, but not cascading the failure.

Reply:
Could we just delete and re-create the collection on index migration?

Reply:
We could, but it's probably overkill. Adding or removing indexes doesn't necessarily mean the data inside is invalidated, just that we've found another/more efficient way to access it.